### DISCUSSION TITLE
Not null assignment operator i C#

### DISCUSSION THREAD

**Discussion by gmjn** [dotnet/csharplang#8280] [Language Ideas]
C# should implement a non-null assignment operator.

Now:
```csharp
if (switch is not null)
{
    switch.Value = true;
}
```

Easier desired way:
```csharp
switch?.Value =? true;
```

**Comment by CyrusNajmabadi** [comment]
@RikkiGibson has a proposal for thsi already.  Syntax woudl simply be `s?.value = true`

**Reply by 333fred** [reply]
https://github.com/dotnet/csharplang/issues/6045

**Reply by gmjn** [reply]
No, the syntax you refer to is wrong.

This works when comparing:
```csharp
s?.value == true
```

This non't work when assigning:
```csharp
s?.value = true
```

**Reply by CyrusNajmabadi** [reply]
> No, the syntax you refer to is wrong.

No clue what you mean by that.  The linked proposal is about making the latter syntax work.  Which will then have exactly the same semantics as what you asked for in your original post.  this is not about `==`.  THis is about assignment.

**Reply by ParsaMehdipour** [reply]
> No, the syntax you refer to is wrong.
> 
> This works when comparing:
> 
> ```cs
> s?.value == true
> ```
> 
> This non't work when assigning:
> 
> ```cs
> s?.value = true
> ```

Well your question was about non-null assignment which I believe the following code would do just the same:

```
c?.obj = new object();
```

Or in your situation:

```
s?.value = true
```

**Reply by gmjn** [reply]
Yes, I thought https://github.com/dotnet/csharplang/issues/6045 was the duplicate request. But in issue it should be possible to assign values ​​to null values. I want a new operator so that it will only be assigned when value is not null. C# is complicated and there are many issues that I don't know about. This post was just to inform the C# team that I see the possibility of =? operator. Maybe a bad idea.

**Reply by KennethHoff** [reply]
> I want a new operator so that it will only be assigned when value is not null.

You can't assign values to null

**Reply by CyrusNajmabadi** [reply]
> I want a new operator so that it will only be assigned when value is not null. 

Yes, that's what https://github.com/dotnet/csharplang/issues/6045 is.
